src/main/java/org/cryptomator/common/locationpresets/GoogleDriveMacLocationPresetsProvider.java

```diff
@@ -0,0 +1,147 @@
+	 *
+	 * @return {@code true} if a matching root location is present, otherwise {@code false}.
+	 * @throws UncheckedIOException if an I/O error occurs during directory traversal.
+	 */
+	public static boolean isRootLocationPresent() {
+		try (var dirStream = Files.list(ROOT_LOCATION)) {
+			return dirStream.anyMatch(path -> Files.isDirectory(path) && PATTERN.test(path.getFileName().toString()));
+		} catch (IOException | UncheckedIOException e) {
+			return false;
+		}
+	}
+
+	/**
+	 * Returns Google Drive preset String.
+	 *
+	 * @param accountPath The path to the Google Drive account directory (e.g. {@code ~/Library/CloudStorage/GoogleDrive-username})
+	 * @return {@code String}. For example: "Google Drive - username"
+	 */
+	private String getDriveLocationString(Path accountPath, Path drivePath) {
+		String accountName = accountPath.getFileName().toString().replace("GoogleDrive-", "");
+		return STR."Google Drive - \{accountName}";
+	}
+
+	/**
+	 * Retrieves a stream of `LocationPreset` objects from directories within the `ROOT_LOCATION` path.
+	 * <p>
+	 * This method lists all directories in the `ROOT_LOCATION`, filters them based on whether their names match
+	 * a predefined pattern (`PATTERN`), and then extracts presets using `getPresetsFromAccountPath(Path)`.
+	 * The results are collected into a list and returned as a stream.
+	 * <p>
+	 *
+	 * @return a stream of `LocationPreset` objects representing valid cloud storage directory locations.
+	 */
+	private Stream<LocationPreset> getCloudStorageDirLocations() {
+		try (var dirStream = Files.list(ROOT_LOCATION)) {
+			return dirStream.filter(path -> Files.isDirectory(path) && PATTERN.test(path.getFileName().toString())).flatMap(this::getPresetsFromAccountPath).toList().stream();
+		} catch (IOException | UncheckedIOException e) {
+			return Stream.empty();
+		}
+	}
+
+	/**
+	 * Retrieves a stream of `LocationPreset` objects from a given Google Drive account path.
+	 * <p>
+	 * This method lists all directories within the provided `accountPath` and filters them
+	 * to identify folders whose names match any of the translations defined in `MY_DRIVE_TRANSLATIONS`.
+	 * <p>
+	 * Each matching folder is then converted into a `LocationPreset` object.
+	 *
+	 * @param accountPath the root path of the Google Drive account to scan.
+	 * @return a stream of `LocationPreset` objects representing matching directories.
+	 */
+	private Stream<LocationPreset> getPresetsFromAccountPath(Path accountPath) {
+		try (var driveStream = Files.list(accountPath)) {
+			List<Path> directories = StreamSupport.stream(driveStream.spliterator(), false).toList();
+			return directories.stream().filter(preset -> MY_DRIVE_TRANSLATIONS.stream().anyMatch(translation -> preset.getFileName().toString().equalsIgnoreCase(translation))).map(drivePath -> new LocationPreset(getDriveLocationString(accountPath, drivePath), drivePath));
```

Comment:
Instead of streaming MY_DRIVE_TRANSLATIONS, simply use `contains()` method of the `Collection` interface.

And please split this over multiple lines.